src/stores/movies.js

```diff
@@ -0,0 +1,28 @@
+import { defineStore } from "pinia";
+import useMoviesApi from "@/api/useMoviesApi.js";
+const { retrieveMovies } = useMoviesApi();
+
+export const useMovieStore = defineStore({
+  id: "movies",
+  state: () => ({
+    movies: [],
+  }),
+  getters: {},
+  actions: {
+    async getMovies() {
+      const res = await retrieveMovies();
+      this.movies = res.map((movie) => {
+        const { length, genre, poster_url } = movie;
+        const hours = Math.floor(length / 60);
+        const minutes = `0${length % 60}`.slice(-2);
+        const lengthInString = `${hours}h ${minutes}min`;
+        return {
+          ...movie,
```

Comment:
I'm not sure if such behavior is wanted but if you would like to omit destructured properties from `movie` you can declare in 15 line something like `const { length, genre, poster_url, ...restMovie } = movie;` and then spread here this `restMovie` instead of `movie` 😄 but if `genre` and `poster_url` are expected (or just can exist because they don't break anything) to be in returned `movie` it's cool as it is 👌

Reply:
They don't break anything, at least for now. Some of them will actually be useful, e.g. genre.id in the future.